### PR TITLE
Update Zlib, LibPNG, ImageMagick deps

### PR DESCRIPTION
**Pull request by IanButterworth**
Passes locally, but has a pretty big rewrite of `build.jl` so needs a review.

Relates to:
https://github.com/JuliaIO/ImageMagickBuilder/pull/9
https://github.com/bicycle1885/ZlibBuilder/pull/14

### CODE REVIEW

File: deps/build.jl [hunk 1]
```diff
@@ -1,93 +1,45 @@
...
-        jpeg.check_deps()
-        tiff.check_deps()
-        im.check_deps()
-    end
-    """)
+                
+open("deps.jl", "a") do io
+    write(io, "libversion() =  $(repr(version))")
 end
+
```
Comment (giordano) [inline deps/build.jl]: Line non terminated by new line ☺️

**Review by giordano — COMMENTED** [review]
Could you please add the cirrus script from my other pull request? I did this to see if freebsd works with the new dependencies

File: deps/build.jl (hunk 1 above)
Comment (IanButterworth) [reply deps/build.jl]: 👍🏻

**Comment by IanButterworth** [comment]
@giordano Done! Thanks

**Comment by IanButterworth** [comment]
Currently waiting on merging of https://github.com/fhs/ZipFile.jl/pull/54 (and a new ZipFile version release)

**Comment by SimonDanisch** [comment]
Looks almost perfect :) I don't really know where those failures on nightly & 1.2 come from... Locally, the Julia version doesn't seem to make a difference...

**Review by SimonDanisch — APPROVED** [review]

**Comment by giordano** [comment]
> I don't really know where those failures on nightly & 1.2 come from

Do you mean the failures on FreeBSD?

**Comment by SimonDanisch** [comment]
Yup! Since 1.0 & 1.1 pass on FreeBSD!

**Comment by SimonDanisch** [comment]
And I don't know why Colors.jl should fail to compile on FreeBSD, when it compiles fine on linux with all my Julia versions!

**Comment by giordano** [comment]
I think Julia on FreeBSD has some issues, like https://github.com/JuliaLang/julia/issues/31559, which perhaps is not related to this one, though

**Comment by SimonDanisch** [comment]
Ah well, I'd say we could just merge :D

**Comment by giordano** [comment]
Yes, in this situation I think that passing on Julia 1.0 and 1.1 on FreeBSD is sufficient

**Comment by SimonDanisch** [comment]
@timholy I can tag a new version, but let me know if you want to give this a spin and tag yourself! :)

**Comment by timholy** [comment]
Go for it!

**Pull request merged**